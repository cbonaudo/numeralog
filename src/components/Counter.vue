<template>
  <div class="hello">
    <div class="counter_sidebar">
      <button>Reset</button><br /><button>Random Number</button><br /><button>Change Color</button>
    </div>
    <h1>Counter</h1>
    <div>
      <span>
        <button v-for="(figure, index) in figures" :key="index" v-on:click="increment(index);">
          ^
        </button>
      </span>
    </div>
    <div class="counter">
      <span v-for="(figure, index) in figures" :key="index">{{ figure }} </span>
    </div>
    <div>
      <span>
        <button v-for="(figure, index) in figures" :key="index" v-on:click="decrement(index);">
          v
        </button>
      </span>
    </div>
    <div>
      <p class="translatedNumber">{{ translatedNumber }}</p>
    </div>
  </div>
</template>

<script>
import { translateNumber } from '../utils/translation';

export default {
  name: 'Counter',
  data() {
    return {
      figures: [0, 0, 0, 1],
      translatedNumber: 'Un',
      // Increment or go back to 0 if figure is already at 9
      increment(index) {
        let incremented = this.figures[index];
        if (this.figures[index] === 9) {
          incremented = 0;
        } else {
          incremented += 1;
        }
        this.figures.splice(index, 1, incremented);
        this.translatedNumber = translateNumber(this.figures);
      },
      // Decrement or go back to 9 if figure is already at 0
      decrement(index) {
        let decremented = this.figures[index];
        if (this.figures[index] === 0) {
          decremented = 9;
        } else {
          decremented -= 1;
        }
        this.figures.splice(index, 1, decremented);
        this.translatedNumber = translateNumber(this.figures);
      },
    };
  },
};
</script>

<style scoped>
.counter_sidebar {
  background-color: green;
  padding: 10px;
  height: 200px;
  position: absolute;
}
</style>
