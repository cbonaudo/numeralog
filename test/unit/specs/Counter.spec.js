import Vue from 'vue';
import Counter from '@/components/Counter';

describe('Counter.vue', () => {
  const Constructor = Vue.extend(Counter);
  const vm = new Constructor().$mount();

  it('should render Counter title', () => {
    expect(vm.$el.querySelector('h1').textContent).toEqual('Counter');
  });

  it('should render Un as starting translatedNumber', () => {
    expect(vm.$el.querySelector('.translatedNumber').textContent).toEqual('Un');
  });

  it('should render 0 0 0 1 as starting counter', () => {
    expect(vm.$el.querySelector('.counter').textContent).toEqual('0 0 0 1 ');
  });
});
