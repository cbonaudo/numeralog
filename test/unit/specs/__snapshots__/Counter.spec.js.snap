// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue should match snapshot 1`] = `
<div
  class="hello"
>
  <div
    class="counter_sidebar"
  >
    <button>
      Reset
    </button>
    <br />
    <button>
      Random Number
    </button>
    <br />
    <button>
      Change Color
    </button>
  </div>
   
  <h1>
    Counter
  </h1>
   
  <div>
    <p
      class="translatedNumber"
    >
      Un
    </p>
  </div>
   
  <div>
    <span>
      <button>
        
        ^
      
      </button>
      <button>
        
        ^
      
      </button>
      <button>
        
        ^
      
      </button>
      <button>
        
        ^
      
      </button>
    </span>
  </div>
   
  <div
    class="counter"
  >
    <span>
      0 
    </span>
    <span>
      0 
    </span>
    <span>
      0 
    </span>
    <span>
      1 
    </span>
  </div>
   
  <div>
    <span>
      <button>
        
        v
      
      </button>
      <button>
        
        v
      
      </button>
      <button>
        
        v
      
      </button>
      <button>
        
        v
      
      </button>
    </span>
  </div>
</div>
`;
